docs/developer_guide/custom_filters.md

```diff
@@ -0,0 +1,341 @@
+   - **StripBasePathGatewayFilter**: Removes base path prefixes
+   - **RouteProfileGatewayFilter**: Enables routes based on active Spring profiles
+   - **CookieAffinityGatewayFilter**: Modifies cookie paths for service sharing
+9. **AddSecHeadersGatewayFilter**: Adds user and organization headers for backend services (runs after both GlobalUriFilter and any application-specific filters)
+10. **Proxy Filter**: Handles the actual HTTP request to the backend
+
+Note that the GlobalUriFilter can either flow directly to AddSecHeadersGatewayFilter or through application-specific filters first, depending on the route configuration.
+
+## Main Custom Filters
+
+### RemoveSecurityHeadersGatewayFilter
+
+Removes security headers from incoming requests to prevent impersonation.
+
+**Class**: `org.georchestra.gateway.filter.headers.RemoveSecurityHeadersGatewayFilterFactory`
+
+**Configuration**:
+```yaml
+spring:
+  cloud:
+    gateway:
+      default-filters:
+        - RemoveSecurityHeaders
+```
+
+### ResolveTargetGlobalFilter
+
+Resolves the target service for the request.
+
+**Class**: `org.georchestra.gateway.filter.global.ResolveTargetGlobalFilter`
+
+**Configuration**: This filter is automatically added to the filter chain.
+
+### ResolveGeorchestraUserGlobalFilter
+
+Resolves the geOrchestra user from the security context.
+
+**Class**: `org.georchestra.gateway.security.ResolveGeorchestraUserGlobalFilter`
+
+**Configuration**: This filter is automatically added to the filter chain.
+
+### AddSecHeadersGatewayFilter
+
+Adds security headers to outgoing requests to provide user information to backend services. This filter works with multiple header contributors to populate different types of headers.
+
+**Class**: `org.georchestra.gateway.filter.headers.AddSecHeadersGatewayFilterFactory`
+
+**Header Contributors**:
+- `GeorchestraUserHeadersContributor`: Adds user-related headers (username, email, roles)
+- `GeorchestraOrganizationHeadersContributor`: Adds organization-related headers
+- `JsonPayloadHeadersContributor`: Adds Base64-encoded JSON payloads with user and org information
+- `SecProxyHeaderContributor`: Adds the sec-proxy header when enabled
+
+**Configuration**:
+```yaml
+spring:
+  cloud:
+    gateway:
+      default-filters:
+        - AddSecHeaders
+```
+
+## Other Custom Filters
+
+### LoginParamRedirectGatewayFilter
+
+Redirects to login page when the `login` query parameter is present.
+
+**Class**: `org.georchestra.gateway.filter.global.LoginParamRedirectGatewayFilterFactory`
+
+**Configuration**:
+```yaml
+spring:
+  cloud:
+    gateway:
+      default-filters:
+        - LoginParamRedirect
+```
+
+### ApplicationErrorGatewayFilter
+
+Handles application errors and provides custom error responses.
+
+**Class**: `org.georchestra.gateway.filter.global.ApplicationErrorGatewayFilterFactory`
+
+**Configuration**:
+```yaml
+spring:
+  cloud:
+    gateway:
+      default-filters:
+        - ApplicationError
+```
+
```

Comment:
please explain how the custom error response is provided (custom error template ?)